doc/source/config.rst

```diff
@@ -0,0 +1,376 @@
+          std::cout << "\"" << p.first << "\" : \"" << p.second << "\"\n";
+        }
+
+The above produces the following output. Observe that the prefix
+is *stripped* from the retrieved parameter names.
+
+.. code-block:: bash
+
+   VFS S3 settings:
+   "connect_max_tries" : "5"
+   "connect_scale_factor" : "25"
+   "connect_timeout_ms" : "3000"
+   "endpoint_override" : ""
+   "max_parallel_ops" : "8"
+   "multipart_part_size" : "5242880"
+   "proxy_host" : ""
+   "proxy_password" : ""
+   "proxy_port" : "0"
+   "proxy_scheme" : "https"
+   "proxy_username" : ""
+   "region" : "us-east-1"
+   "request_timeout_ms" : "3000"
+   "scheme" : "https"
+   "use_virtual_addressing" : "true"
+
+Saving/Loading config to/from file
+----------------------------------
+
+You can save the configuration parameters you used in your program
+into a (local) text file, and subsequently load them from the
+file into a new TileDB config if needed as follows:
+
+.. content-tabs::
+
+   .. tab-container:: cpp
+      :title: C++
+
+      .. code-block:: c++
+
+        // Save to file
+        Config config;
+        config["sm.tile_cache_size"] = 0;
+        config.save_to_file("tiledb_config.txt");
+
+        // Load from file
+        Config config_load("tiledb_config.txt");
+        std::string tile_cache_size = config_load["sm.tile_cache_size"];
+        std::cout << "\nTile cache size after loading from file: " << tile_cache_size
+                  << "\n";
+
+The above code creates a config object, changes the tile cache size to ``0``,
+and saves the entire configuration into a file. Next, it creates a new
+config loading the values from the created file. Running the program
+produces the following output. Observe that the loaded tile cache size
+value is ``0``, which is the value we altered prior to saving the config
+to the file.
+
+.. code-block:: bash
+
+   Tile cache size after loading from file: 0
+
+Inspecting the contents of the exported config file, we get the following:
+
+.. code-block:: bash
+
+  $ cat tiledb_config.txt
+  sm.array_schema_cache_size 10000000
+  sm.check_coord_dups true
+  sm.dedup_coords false
+  sm.enable_signal_handlers true
+  sm.fragment_metadata_cache_size 10000000
+  sm.num_async_threads 1
+  sm.num_tbb_threads -1
+  sm.tile_cache_size 0
+  vfs.file.max_parallel_ops 8
+  vfs.min_parallel_size 10485760
+  vfs.num_threads 8
+  vfs.s3.connect_max_tries 5
+  vfs.s3.connect_scale_factor 25
+  vfs.s3.connect_timeout_ms 3000
+  vfs.s3.max_parallel_ops 8
+  vfs.s3.multipart_part_size 5242880
+  vfs.s3.proxy_port 0
+  vfs.s3.proxy_scheme https
+  vfs.s3.region us-east-1
+  vfs.s3.request_timeout_ms 3000
+  vfs.s3.scheme https
+  vfs.s3.use_virtual_addressing true
+
+Note that ``vfs.s3.proxy_host``, ``vfs.s3.proxy_username`` and
+``vfs.s3.proxy_password`` are not exported for security purposes.
+
+Summary of Parameters
+---------------------
+
+Below we provide a table with all the TileDB configuration parameters,
```

Comment:
Do we want to just link to the C or C++ API reference here, instead of embedding the whole table?